distributor.go

```diff
@@ -53,8 +53,8 @@ type Distributor struct {
 type ReadRing interface {
 	prometheus.Collector
 
-	Get(key uint32) (ring.IngesterDesc, error)
-	GetAll() []ring.IngesterDesc
+	Get(key uint32, n int, heartbeatTimeout time.Duration) ([]ring.IngesterDesc, error)
+	GetAll(heartbeatTimeout time.Duration) []ring.IngesterDesc
```

Comment:
Another approach might be to exclude the `heartbeatTimeout` from the parameters and instead implement it as a decorator. Not sure it's a _better_ approach, mind you.